funnel/models/commentvote.py

```diff
@@ -113,6 +113,30 @@ def __init__(self, **kwargs):
         super(Commentset, self).__init__(**kwargs)
         self.count = 0
 
+    @property
+    def parent(self):
+        parent = None  # project or proposal object
+        if self.project is not None:
+            parent = self.project
+        elif self.proposal is not None:
+            parent = self.proposal
+        return parent
+
+    @property
+    def parent_commentset_url(self):
+        url = None  # project or proposal object
+        if self.project is not None:
+            url = self.project.url_for('comments', _external=True)
+        elif self.proposal is not None:
+            url = self.proposal.url_for(_external=True)
+        return url
+
+    def permissions(self, user, inherited=None):
+        perms = super().permissions(user, inherited)
+        if user is not None:
+            perms.update({'new_comment', 'vote_comment'})
```

Comment:
I did the timeit test and it's faster to call `perms.add` up to three times. Lower overhead than constructing a set (or even tuple) and giving it to `perms.update`.